File: apps/mobile/ios/Visvine/Models/Feed.swift
import Foundation

/// One channel the feed draws from — mirrors lib/messages/shared/feed.ts `FeedPlace`.
struct NamedRef: Codable, Hashable {
    let id: String
    var name: String
}

struct FeedPlace: Codable {
    let conversationId: String
    var channel: NamedRef
    var space: NamedRef
}

/// A post and its comments — mirrors `FeedPost`. The message is the same
/// `SerializedMessage` subset the conversations screens decode.
struct FeedPost: Codable, Identifiable {
    var id: String { message.id }
    let conversationId: String
    var channel: NamedRef
    var space: NamedRef
    var message: Message
    var comments: [Message]
}

/// GET /api/feed?spaceId=&cursor=&limit= — mirrors `FeedPage`.
struct FeedPage: Codable {
    var posts: [FeedPost]
    var nextCursor: String?
    /// The channels the viewer may post into. First page only.
    var targets: [FeedPlace]?
}

File: apps/mobile/android/app/src/main/java/com/visvine/mobile/data/model/Feed.kt
package com.visvine.mobile.data.model

import kotlinx.serialization.Serializable

/** An `{ id, name }` reference — a channel or a space, named for a row. */
@Serializable
data class NamedRef(
    val id: String,
    val name: String = "",
)

/** Where a post was written — mirrors lib/messages/shared/feed.ts `FeedPlace`. */
@Serializable
data class FeedPlace(
    val conversationId: String,
    val channel: NamedRef = NamedRef(id = ""),
    val space: NamedRef = NamedRef(id = ""),
)

/** A post and its comments — mirrors `FeedPost`. */
@Serializable
data class FeedPost(
    val conversationId: String,
    val channel: NamedRef = NamedRef(id = ""),
    val space: NamedRef = NamedRef(id = ""),
    val message: Message,
    val comments: List<Message> = emptyList(),
)

/** GET /api/feed → keyset page, newest first; `targets` on the first page only. */
@Serializable
data class FeedPage(
    val posts: List<FeedPost> = emptyList(),
    val nextCursor: String? = null,
    val targets: List<FeedPlace>? = null,
)
